pkg/tasks/cluster.go

```diff
@@ -0,0 +1,58 @@
+package tasks
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	controlapi "github.com/k8ssandra/cass-operator/apis/control/v1alpha1"
+	k8ssandrataskapi "github.com/k8ssandra/k8ssandra-operator/apis/control/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CreateClusterTask(ctx context.Context, kubeClient client.Client, command controlapi.CassandraCommand, namespace, kcName string, datacenters []string, args *controlapi.JobArguments) (*k8ssandrataskapi.K8ssandraTask, error) {
+	if kcName == "" || namespace == "" {
+		return nil, fmt.Errorf("clusterName and namespace must be specified")
+	}
+
+	generatedName := fmt.Sprintf("%s-%s-%d-%d", kcName, command, time.Now().Unix(), rand.Int31())
+	if len(generatedName) > 63 {
+		generatedName = generatedName[:63]
+	}
```

Comment:
[thought] When the K8ssandraTask controller creates `CassandraTask` instances,  [it uses the naming pattern](https://github.com/k8ssandra/k8ssandra-operator/blob/main/controllers/control/k8ssandratask_controller.go#L310):
```go
k8ssandraTaskName + "-" + dcName
```
So we should account for that. Although we don't necessarily have all the existing datacenter names here, so it's probably better to do it in k8ssandra-operator (there's no length check right now).
WDYT?

Reply:
I think both places will need a cutting function. Even here, I'm wondering if I should rather cut the kcName instead of the randomized parts, since if someone has kcName with length 63, this would always create the same name and we end up with problems. I'll modify this functionality (and I can send a PR to k8ssandra-operator to cut the name there too).

I'd say the randomized part is the most important one (just have to be careful that it doesn't start with a number), we can control the output with ``kubectl get k8ssandratasks`` to include the DC+Cluster information.